File: adk/csharp/LiateAgent.cs
using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Net.Http.Headers;
using System.Text;
using System.Text.Json;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Liate.ADK
{
    public class ToolServer
    {
        [JsonPropertyName("command")]
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string? Command { get; set; }

        [JsonPropertyName("args")]
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public List<string>? Args { get; set; }

        [JsonPropertyName("url")]
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string? Url { get; set; }
    }

    public class ActionSpec
    {
        [JsonPropertyName("name")]
        public string Name { get; set; } = "default";

        [JsonPropertyName("intent")]
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string? Intent { get; set; }

        [JsonPropertyName("skills")]
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string? Skills { get; set; }
    }

    public class IdentitySpec
    {
        [JsonPropertyName("memory")]
        public string Memory { get; set; } = "session_csharp_live";
    }

    public class Pillars
    {
        [JsonPropertyName("L")]
        public string L { get; set; } = "sarvam/sarvam-105b";

        [JsonPropertyName("I")]
        public IdentitySpec I { get; set; } = new IdentitySpec();

        [JsonPropertyName("A")]
        public ActionSpec A { get; set; } = new ActionSpec();

        [JsonPropertyName("T")]
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public Dictionary<string, ToolServer>? T { get; set; }

        [JsonPropertyName("E")]
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public Dictionary<string, string>? E { get; set; }
    }

    public class LiateAgent
    {
        public Pillars Spec { get; set; }
        public string Endpoint { get; set; }
        public string? ApiKey { get; set; }

        private static readonly HttpClient _httpClient = new HttpClient();

        public LiateAgent(Pillars spec, string? endpoint = null, string? apiKey = null)
        {
            Spec = spec;
            Endpoint = endpoint 
                ?? Environment.GetEnvironmentVariable("LIATE_ENDPOINT") 
                ?? Environment.GetEnvironmentVariable("LIATE_BASE_URL") 
                ?? "http://localhost:7071";
            ApiKey = apiKey ?? Environment.GetEnvironmentVariable("LIATE_API_KEY");
        }

        public async Task<string> RunAsync(string prompt)
        {
            var payload = new
            {
                spec = Spec,
                prompt = prompt
            };

            string jsonString = JsonSerializer.Serialize(payload);
            string baseUri = Endpoint.TrimEnd('/');
            string url = $"{baseUri}/lapi/v1/{Spec.A.Name}/run";

            using var request = new HttpRequestMessage(HttpMethod.Post, url)
            {
                Content = new StringContent(jsonString, Encoding.UTF8, "application/json")
            };

            if (!string.IsNullOrEmpty(ApiKey))
            {
                request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", ApiKey);
            }

            HttpResponseMessage response;
            try
            {
                response = await _httpClient.SendAsync(request);
            }
            catch
            {
                // Fallback to legacy endpoint
                string fallbackUrl = $"{baseUri}/api/agent/run";
                using var fallbackRequest = new HttpRequestMessage(HttpMethod.Post, fallbackUrl)
                {
                    Content = new StringContent(jsonString, Encoding.UTF8, "application/json")
                };
                if (!string.IsNullOrEmpty(ApiKey))
                {
                    fallbackRequest.Headers.Authorization = new AuthenticationHeaderValue("Bearer", ApiKey);
                }
                response = await _httpClient.SendAsync(fallbackRequest);
            }

            string responseBody = await response.Content.ReadAsStringAsync();

            if (!response.IsSuccessStatusCode)
            {
                throw new Exception($"[Liate ADK Error {(int)response.StatusCode}]: {responseBody}");
            }

            try
            {
                using var doc = JsonDocument.Parse(responseBody);
                var root = doc.RootElement;
                if (root.TryGetProperty("response", out var respProp)) return respProp.GetString() ?? responseBody;
                if (root.TryGetProperty("output", out var outProp)) return outProp.GetString() ?? responseBody;
                if (root.TryGetProperty("result", out var resProp)) return resProp.GetString() ?? responseBody;
            }
            catch
            {
                // Fallback to raw string if not JSON
            }

            return responseBody;
        }
    }
}

File: adk/php/LiateAgent.php
<?php

namespace Liate\ADK;

class LiateAgent
{
    private array $spec;
    private string $endpoint;
    private ?string $apiKey;

    public function __construct(
        array $spec,
        ?string $endpoint = null,
        ?string $apiKey = null
    ) {
        $this->spec = $spec;
        $this->endpoint = $endpoint 
            ?? getenv('LIATE_ENDPOINT') 
            ?: (getenv('LIATE_BASE_URL') ?: 'http://localhost:7071');
        $this->apiKey = $apiKey ?? getenv('LIATE_API_KEY') ?: null;
    }

    public function run(string $prompt): string
    {
        $payload = json_encode([
            'spec' => $this->spec,
            'prompt' => $prompt
        ]);

        $agentName = $this->spec['A']['name'] ?? 'default';
        $base = rtrim($this->endpoint, '/');
        $url = "{$base}/lapi/v1/{$agentName}/run";

        $headers = ['Content-Type: application/json'];
        if ($this->apiKey) {
            $headers[] = "Authorization: Bearer {$this->apiKey}";
        }

        $ch = curl_init($url);
        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
        curl_setopt($ch, CURLOPT_POST, true);
        curl_setopt($ch, CURLOPT_POSTFIELDS, $payload);
        curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
        curl_setopt($ch, CURLOPT_TIMEOUT, 60);

        $response = curl_exec($ch);
        $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);

        if ($httpCode >= 400 || $response === false) {
            // Fallback to /api/agent/run
            $fallbackUrl = "{$base}/api/agent/run";
            curl_setopt($ch, CURLOPT_URL, $fallbackUrl);
            $response = curl_exec($ch);
            $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
        }

        curl_close($ch);

        if ($httpCode >= 400) {
            throw new \RuntimeException("[Liate ADK Error {$httpCode}]: {$response}");
        }

        $data = json_decode($response, true);
        if (is_array($data)) {
            return $data['response'] ?? $data['output'] ?? $data['result'] ?? $response;
        }

        return (string) $response;
    }
}
